src/components/ShareTrajectoryModal/index.tsx

```diff
@@ -62,62 +68,46 @@ const ShareTrajectoryModal = ({
             timeValue =
                 displayTimes.roundedLastFrameTime -
                 displayTimes.roundedTimeStep;
         } else {
-            timeValue = roundToTimestepPrecision(
+            timeValue = roundToTimeStepPrecision(
                 inputAsNumber,
                 displayTimes.roundedTimeStep
             );
         }
-        setLastEnteredNumber(timeValue);
-        setLinkUrl(
-            editUrlParams(
-                window.location.href,
-                timeValue.toString(),
-                URL_PARAM_KEY_TIME
-            )
-        );
-    };
-
-    const [timeRadioGroupValue, setTimeRadioGroupValue] = React.useState(1);
-    const [allowTimeInput, setAllowTimeInput] = React.useState(true);
-    const currentTime = roundToTimestepPrecision(
-        displayTimes.roundedTime,
-        displayTimes.roundedTimeStep
-    );
-    const [linkUrl, setLinkUrl] = React.useState(
-        editUrlParams(
-            window.location.href,
-            currentTime.toString(),
-            URL_PARAM_KEY_TIME
-        )
-    );
-    const [lastEnteredNumber, setLastEnteredNumber] =
-        React.useState(currentTime);
-    const onRadioChange = (e: RadioChangeEvent) => {
-        setTimeRadioGroupValue(e.target.value);
-        handleAllowTimeInput();
+        setUserEnteredTime(timeValue);
     };
 
     const UrlLinkPanel: JSX.Element = (
         <>
             <VerticalFlexbox gap={2}>
                 <Radio.Group
                     className={styles.customRadioGroup}
-                    value={timeRadioGroupValue}
-                    onChange={onRadioChange}
+                    value={startTimeToUse}
+                    onChange={(e: RadioChangeEvent) => {
+                        setStartTimeToUse(e.target.value);
+                    }}
                 >
                     <Space direction="vertical">
-                        <Radio value={1}>From beginning</Radio>
-                        <Radio value={2}>
+                        <Radio value={TimeToUse.BEGINNING}>
+                            From beginning
+                        </Radio>
+                        <Radio value={TimeToUse.USER_ENTERED}>
                             Start at{" "}
                             <Input
-                                className={styles.timeInput}
-                                disabled={allowTimeInput}
+                                className={classNames(styles.timeInput, {
+                                    [styles.disabled]:
+                                        startTimeToUse === TimeToUse.BEGINNING,
+                                })}
                                 defaultValue={currentTime}
                                 onChange={handleTimeInput}
+                                value={userEnteredTime}
+                                onClick={() => {
+                                    setStartTimeToUse(TimeToUse.USER_ENTERED);
+                                }}
```

Comment:
this is the thing I say to ask UX about: but it made sense to me to be  able to just click into this input box